src/expressions/functions/string.ts

```diff
@@ -159,13 +160,46 @@ export function toUpper(original: Expr): CypherFunction {
     return new CypherFunction("toUpper", [original]);
 }
 
+/** Implements a trim function with a trim expression `trim(BOTH 'x' FROM 'xxxhelloxxx')` */
+class TrimFunction extends CypherFunction {
+    private readonly typeOrInput: "BOTH" | "LEADING" | "TRAILING";
+    private readonly trimChar: Expr;
+    private readonly input: Expr;
+
+    constructor(typeOrInput: "BOTH" | "LEADING" | "TRAILING", trimChar: Expr, input: Expr) {
```

Comment:
```suggestion
    constructor(typeOrInput: "BOTH" | "LEADING" | "TRAILING", trimChars: Expr, input: Expr) {
```
Nitpick: I would put chars as plural to reflect better the usage